routes/post.go

```diff
@@ -1203,13 +1191,153 @@ func (fes *APIServer) GetSinglePost(ww http.ResponseWriter, req *http.Request) {
+		return nil, err
+	}
+
+	// Create a map of all the profile pub keys associated with our posts + comments.
+	profilePubKeyMap := make(map[lib.PkMapKey][]byte)
+	profilePubKeyMap[lib.MakePkMapKey(posterPublicKeyBytes)] = posterPublicKeyBytes
+
+	// Determine whether or not the posters of the "single post" we are fetching is blocked by the reader.  If the
+	// poster of the single post is blocked, we will want to include the single post, but not any of the comments
+	// created by the poster that are children of this "single post".
+	_, isCurrentPosterBlocked := blockedPublicKeys[lib.PkToString(posterPublicKeyBytes, fes.Params)]
 	for _, commentEntry := range commentEntries {
+		pkMapKey := lib.MakePkMapKey(commentEntry.PosterPublicKey)
+		// Remove comments that are blocked by either the reader or the poster of the root post
+		if _, ok := blockedPublicKeys[lib.PkToString(commentEntry.PosterPublicKey, fes.Params)]; !ok && profilePubKeyMap[pkMapKey] == nil {
+			profilePubKeyMap[pkMapKey] = commentEntry.PosterPublicKey
+		}
+	}
+
+	// Filter out restricted PosterPublicKeys.
+	filteredProfilePubKeyMap, err := fes.FilterOutRestrictedPubKeysFromMap(
+		profilePubKeyMap, readerPublicKeyBytes, "leaderboard" /*moderationType*/, utxoView)
+	if err != nil {
+		return nil, err
+	}
+
+
+	// Figure out if the current poster is greylisted.  If the current poster is greylisted, we will add their
+	// public key back to the filteredProfileMap so their profile will appear for the current post
+	// and any parents, but we will remove any comments by this greylisted user.
+	isCurrentPosterGreylisted := false
+	if _, ok := filteredProfilePubKeyMap[lib.MakePkMapKey(posterPublicKeyBytes)]; !ok {
+		// Get the userMetadata for the currentPosters
+		currentPosterUserMetadataKey := append([]byte{}, _GlobalStatePrefixPublicKeyToUserMetadata...)
+		currentPosterUserMetadataKey = append(currentPosterUserMetadataKey, posterPublicKeyBytes...)
+		var currentPosterUserMetadataBytes []byte
+		currentPosterUserMetadataBytes, err = fes.GlobalState.Get(currentPosterUserMetadataKey)
+		if err != nil {
+			return nil, err
+		}
+		// If the currentPoster's userMetadata doesn't exist, then they are no greylisted, so we can exit.
+		if currentPosterUserMetadataBytes != nil {
+			// Decode the currentPoster's userMetadata.
+			currentPosterUserMetadata := UserMetadata{}
+			err = gob.NewDecoder(bytes.NewReader(currentPosterUserMetadataBytes)).Decode(&currentPosterUserMetadata)
+			if err != nil {
+				return nil, err
+			}
+			// If the currentPoster is not blacklisted (removed everywhere) and is greylisted (removed from leaderboard)
+			// add them back to the filteredProfilePubKeyMap and note that the currentPoster is greylisted.
+			if currentPosterUserMetadata.RemoveFromLeaderboard && !currentPosterUserMetadata.RemoveEverywhere {
+				isCurrentPosterGreylisted = true
+				filteredProfilePubKeyMap[lib.MakePkMapKey(posterPublicKeyBytes)] = posterPublicKeyBytes
+			}
+		}
+	}
```

Comment:
```suggestion
        posterPKID := utxoView.GetPKIDForPublicKey(posterPublicKeyBytes)
	isCurrentPosterGreylisted := false
	if fes.IsUserGraylisted(posterPKID.PKID) {
		isCurrentPosterGreylisted = true
		filteredProfilePubKeyMap[lib.MakePkMapKey(posterPublicKeyBytes)] = posterPublicKeyBytes
	}
```

and I think we replicate this bit of logic above in the original endpoint function

Reply:
actually do we need to check graylist status of each commenter? I think this logic was specifically put in for the poster of the post hash hex from the request data. If we don't need it here, we should kill it here. but may as well move the cleaner graylisted logic to the earlier bit if we can.

Reply:
This is a good point - this is no longer needed. I've updated accordingly.